projects/user_publication/gscholar.py

```diff
@@ -120,33 +124,83 @@ def get_authors(self, pub):
         Returns:
             list: list of authors in ["firstname lastname",] format
         """
-        bibt = pub['bibtex']
-        bibt = utils.decode_unicode_text(bibt)
-        match = re.search(r'author\s*=\s*\{([\w\s-,]+)\}', bibt)
-        if match:
-            authors_match = match.groups()[0]
-        else:
-            # this mostly never happens - if it does - log it
-            logger.info(f"cannot parse bibtex, so ignoring - {bibt}")
-            return
-        authors = authors_match.split(" and ")
+        authors = utils.decode_unicode_text(pub['bib']['author'])
+        # substitute all charachter from text except
+        # ',' , \w word charachter, \s whitespace charachter
+        # - any other charachter
+        # inspired from django.utils.text.slugify
+        authors = re.sub(r"[^\,\w\s-]", "", authors)
+        authors = authors.split(" and ")
         parsed_authors = [utils.parse_author(author) for author in authors]
         return parsed_authors
 
     def get_pub_model(self, pub):
-        return Publication(
+        entry_type = []
+        # bibtex ENTRYTYPE field
+        if 'ENTRYTYPE' in pub['bib']:
+            entry_type.append(pub['bib']['ENTRYTPE'])
+        if 'pub_type' in pub['bib']:
+            entry_type.append(pub['bib']['pub_type'])
+        forum = ''
+        if 'booktitle' in pub['bib']:
+            forum = pub['bib']['booktitle']
+        if 'journal' in pub['bib']:
+            forum = pub['bib']['journal']
+        pub_type = utils.get_pub_type(entry_type, forum)

```

Comment:
I didn't notice this in the previous code review, but there are now two functions called `get_pub_type` that both do almost the exact same thing in the publication utils module. This needs to be cleaned up. The functions either need to be merged or their names need to be changed. They should also be in the same scope (i.e. in the `PublicationUtils` class, since that's where the existing function is).

Reply:
Good catch. Not sure how I missed this.